src/interceptors/XMLHttpRequest/XMLHttpRequestOverride.ts

```diff
@@ -682,5 +609,88 @@ export const createXMLHttpRequestOverride = (
         request.setRequestHeader(headerName, headerValue)
       }
     }
+
+    toPassthroughRequest(): XMLHttpRequest {
+      const request = new XMLHttpRequest()
+
+      request.open(
+        this.method,
+        this.url,
+        this.async ?? true,
+        this.user,
+        this.password
+      )
+
+      request.addEventListener('readystatechange', () => {
+        // Set original response headers as soon as they are received.
+        if (request.readyState === XMLHttpRequest.HEADERS_RECEIVED) {
+          const responseHeaders = request.getAllResponseHeaders()
+          this.log('original response headers (raw):\n', responseHeaders)
+
+          this._responseHeaders = stringToHeaders(responseHeaders)
+
+          this.log(
+            'original response headers (normalized)',
+            this._responseHeaders
+          )
+        }
+      })
+
+      request.addEventListener('progress', () => {
+        const chunk = request.response
+        const nextChunk =
```

Comment:
@avm99963, so the "Invalid state" error was caused by the interceptor accessing `request.responseText` of the original request when its `responseType` value is anything else but `"text"`. Turns out, that's a no-op (I'd expect `responseText` to return a text representation of the response body but that's not how JSDOM works, at least).

I've added the logic that checks the value of `request.response`, and if it's an array buffer, we skip the encoding part, create a uint8array out of the response buffer, and add it to the internal response buffer representation. That seems to work.

Reply:
Nice catch! So it seems like since we didn't forward `responseType` before, it was always `''`, the default (which is the same as `'text'`), so `responseText` always returned the response text.

LGTM!